Lib/test/test_threading.py

```diff
@@ -1338,6 +1339,22 @@ def run(self):
         # explicitly break the reference cycle to not leak a dangling thread
         thread.exc = None
 
+    def test_multithread_modify_file_noerror(self):
+        # See issue25872
+        def modify_file():
+            with open(os_helper.TESTFN, 'w', encoding='utf-8') as fp:
+                fp.write(' ')
+                traceback.format_stack()
+            self.addCleanup(os_helper.unlink, os_helper.TESTFN)
```

Comment:
We only need addCleanup to be called once, right? So should it be outside of the mofidy_file() function?